keps/sig-api-machinery/1040-priority-and-fairness/README.md

```diff
@@ -2840,143 +2876,271 @@ GA:
 
-* **Are there any tests for feature enablement/disablement?** No. Manual tests
-  will be run before switching feature gate to beta.
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The feature will be restored.
+
+###### Are there any tests for feature enablement/disablement?
+
+No.  Manual tests were run before switching feature gate to beta.
 
 ### Rollout, Upgrade and Rollback Planning
 
-* **How can a rollout fail? Can it impact already running workloads?** A
-  misconfiguration could cause apiserver requests to be rejected, which could
-  have widespread impact such as: (1) rejecting controller requests, thereby
-  bringing a lot of things to a halt, (2) dropping node heartbeats, which may
-  result in overloading other nodes, (3) rejecting kube-proxy requests to
-  apiserver, thereby breaking existing workloads, (4) dropping leader election
-  requests, resulting in HA failure, or any combination of the above.
-
-* **What specific metrics should inform a rollback?** An abnormal spike in the
-  `apiserver_flowcontrol_rejected_requests_total` metric should potentially be
-  viewed as a sign that kube-apiserver is rejecting requests, potentially
-  incorrectly. The `apiserver_flowcontrol_request_queue_length_after_enqueue`
-  metric getting too close to the configured queue length could be a sign of
-  insufficient queue size (or a system overload), which can be precursor to
-  rejected requests.
-
-* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
-  No. Manual tests will be run before switching feature gate to beta.
-
-* **Is the rollout accompanied by any deprecations and/or removals of features, APIs, 
-  fields of API types, flags, etc.?** Yes, `--max-requests-inflights` will be
-  deprecated in favor of APF.
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+A misconfiguration could cause apiserver requests to be rejected,
+which could have widespread impact such as: (1) rejecting controller
+requests, thereby bringing a lot of things to a halt, (2) dropping
+node heartbeats, which may result in overloading other nodes, (3)
+rejecting kube-proxy requests to apiserver, thereby breaking existing
+workloads, (4) dropping leader election requests, resulting in HA
+failure, or any combination of the above.
+
+###### What specific metrics should inform a rollback?
+
+An abnormal spike in the
+`apiserver_flowcontrol_rejected_requests_total` metric should
+potentially be viewed as a sign that kube-apiserver is rejecting
+requests, potentially incorrectly. The
+`apiserver_flowcontrol_request_queue_length_after_enqueue` metric
+getting too close to the configured queue length could be a sign of
+insufficient queue size (or a system overload), which can be precursor
+to rejected requests.
```

Comment:
I think people also need to look at the total requests, otherwise it might be the case that it is correctly rejecting a large surge of requests?

Reply:
Yes that touches on an important point that I am not sure what to say about.  It arises in both this section ("What specific metrics should inform a rollback?") and the "What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?" section.  The current text identifies metrics that indicate an unhappy situation, but it is harder to say how to distinguish whether the problem is in APF or the problem is that the apiserver is honestly overloaded.  Also, in the latter section, I am not sure what "the service" refers to --- is it just APF or the apiserver or a whole cluster or ...?

Reply:
Maybe something like this?

* high latency / long queue lengths + low request volume = probably APF misconfiguration (or maybe bug)
* high latency / long queue lengths + high request volume = probably APF is saving your cluster from an outage

Reality is probably going to be a mix of these, probably a high latency condition will help one identify a configuration that can be optimized somewhere.

I'd take "service" in this case to mean the apiserver I guess -- I don't think the template authors had this situation in mind.

Reply:
> I'd take "service" in this case to mean the apiserver I guess -- I don't think the template authors had this situation in mind.

Yes - the "service" was supposed to be generic enough so that people may accomodate the meaning to their needs/feature.

I like Daniel`s suggestion - we don't have to be perfect in the answer there, we just need to give people overall sense of what they should be looking at.

Just mention the name of the metric they can get request volume from (I think it's called apiserver_requests_total or sth like that, but you should check).